controllers/awsmachinepool_controller_test.go

```diff
@@ -0,0 +1,409 @@
+package controllers_test
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	awsclientupstream "github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capa "sigs.k8s.io/cluster-api-provider-aws/api/v1beta1"
+	expcapa "sigs.k8s.io/cluster-api-provider-aws/exp/api/v1beta1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/giantswarm/capa-iam-operator/controllers"
+	"github.com/giantswarm/capa-iam-operator/pkg/test/mocks"
+)
+
+var _ = Describe("AWSMachinePoolReconciler", func() {
+	var (
+		ctx           context.Context
+		mockCtrl      *gomock.Controller
+		mockIAMClient *mocks.MockIAMAPI
+		reconcileErr  error
+		reconciler    *controllers.AWSMachinePoolReconciler
+		req           ctrl.Request
+		namespace     string
+	)
+
+	SetupNamespaceBeforeAfterEach(&namespace)
+
+	BeforeEach(func() {
+		logger := zap.New(zap.WriteTo(GinkgoWriter))
+		ctx = log.IntoContext(context.Background(), logger)
+
+		mockCtrl = gomock.NewController(GinkgoT())
+
+		ctx := context.TODO()
+
+		mockIAMClient = mocks.NewMockIAMAPI(mockCtrl)
+
+		reconciler = &controllers.AWSMachinePoolReconciler{
+			Client: k8sClient,
+			Log:    ctrl.Log,
+
+			IAMClientAndRegionFactory: func(session awsclientupstream.ConfigProvider) (iamiface.IAMAPI, string) {
+				return mockIAMClient, fakeRegion
+			},
+		}
+
+		err := k8sClient.Create(ctx, &expcapa.AWSMachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"cluster.x-k8s.io/cluster-name": "test-cluster",
+					"cluster.x-k8s.io/watch-filter": "capi",
+				},
+				Name:      "my-awsmp",
+				Namespace: namespace,
+			},
+			Spec: expcapa.AWSMachinePoolSpec{
+				AWSLaunchTemplate: expcapa.AWSLaunchTemplate{
+					IamInstanceProfile: "the-profile",
+				},
+				MaxSize: 3,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		err = k8sClient.Create(ctx, &capa.AWSCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"cluster.x-k8s.io/cluster-name": "test-cluster",
+				},
+				Name:      "my-awsc",
```

Comment:
```suggestion
				Name:      "test-cluster",
```

Reply:
They don't have to, and that's why we have to look up by label. This is strongly on purpose, to avoid wrong assumptions in the code.